### ISSUE TITLE
Put translation strings in multiple files

### ISSUE THREAD

**Issue by Wuzzy2** [CleverRaven/Cataclysm-DDA#23434]
Currently, almost all strings of the game are stored in one *giant* PO. It has over 24,000 strings. This is a big problem as it is not always easy to figure out the context, and the string list in general is pretty chaotic and impossible to navigate. Scrolling over a list of 24,000 strings is not fun, especially not in Transifex.

It would make life a lot easier for translators if the massive amount of strings is at least moved into different PO files. The idea is that breaking down the list of strings in smaller, context-based lists makes it more managable.

I propose the strings be moved in categories like this:

 * Core non-content: All the core strings of the game which is not part of the game content. This includes the main menu and pretty much all of the GUI, but not stuff like item names. Maybe it makes sense to break this category down in sub-categories:
    * Factions
    * Artifacts
    * Main menu
    * Item menu
    * …
 * Core content: Core content such as everything from the Dark Days Ahead content pack might be further broken down into components:
    * Items
    * Bionics
    * Monsters
    * Messages
    * …
 * Mods: The other strings from mods which are not core content should be split by mods. One mod = one resource

This would also make it possible to priotirize resources on Transifex. The core mods and the core GUI are obviously top priority while other non-core mods should get a lower priority. The benefit of this is that translators can more quickly come to an useful result.
Especially the translation of the firearms mods is tedious.

More ideas welcome! :-)

If someone is willing to do this, please do so in a non-destructive way. **No existing translation must get broken because of this.** The transition must be done smoothly. If you can't do this, don't do it at all. Maybe even temporarily lock down Transifex during the transition to make sure no translator work is destroyed.

**Comment by BrettDong** [comment]
The open source game [_0.A.D_ ](https://www.transifex.com/wildfire-games/0ad) is a good example of doing this. Their developers split strings into different categories and put them in separate PO files, such as 'In-Game UI', 'Tutorials', 'Buildings', 'Maps', 'Units'. Someone who is willing to do this could have a look at how they do it technically.

**Comment by Wuzzy2** [comment]
Yes, I was exactly thinking of something along the lines of 0. A.D.

**Comment by nexusmrsep** [comment]
Would it cause the need to translate everything again in Transifex? Even using its own repository of already translated strings it would be no joke to re-link over 24.000 of them, and could revert all teams translation effort to tile one.

**Comment by Wuzzy2** [comment]
Obviously this must be done in a non-destructive manner, otherwise we lose months of work.
Duh!

**Comment by Faalagorn** [comment]
You can search Transifex for occurrence in each json file. I think it applies to C++ files as well.

For example, to search all strings from `construction.json` you can search for `occurrence:lang/json/construction_from_json.py` in Transifex. Compared to actual game files, you just have to replace `data` with `lang` add `_from_json.py` instead of `.json` for it.

**Comment by Wuzzy2** [comment]
Is there any progress on this issue?

The translation resource just continue to grow and grow and grow. It's barely managable anymore.
What's my biggest problem right now that strings from all mods and the base game are combined into a giant master file.

I think splitting off the file on a per-mod basis would be a huge improvement already. I suspect the base game will still be huge but at least you can focus on it now.

It would help translators to priotize their time. It's not good that after you managed to reach 90%, you might still have lots of missing strings in the base game because everything is mixed together …
It is strategically much smarter to first focus on the base game only. I don't know how many strings the mod contribute to the grand total, but I think it's a lot. And since C:DDA has a huge amount of strings, it makes a lot of sense to think about time management.

**Comment by CountAlex** [comment]
I add my voice to this concern. Transifex is painful to use most of the time, and I've spent more time trying to update translation with file upload than I was actually translating.
Separating into categories would at least help to solve this partially and raise quality thanks to better context.
Would like also to mention #39164 - although for me it's about generally unpesponsive and clunky online UI.
And yeah, after a lot of attempts to figure out I still don't know who oversees translation to RU language I work on. Would be nice to have my rights elevated to make reviews on translated entries at least.

**Comment by comradekingu** [comment]
I think moving to Weblate and then tagging up the strings with different labels would help, short of dividing up files.

Anything you do on Transifex removes translator credits, which makes that mess of a platform even harder to handle.

Translated a 16k line document by hand once, 24k is just on another planet. ;)

**Comment by Faalagorn** [comment]
It would, as suggested, also help to split mods at least. These would easy up translating the base game content as a priority for new languages, or the mods in some order.
